### DISCUSSION TITLE
Soumallya Dev - Github Externship Proposal

### DISCUSSION THREAD

**Discussion by soumallyadev07** [hoppscotch/hoppscotch#1666] [Q&A]
**Application number:** 20-05_Sou340_tfa_258

**Full name:** Soumallya Dev

**E-mail:** dev.soumallya@gmail.com

**GitHub username:** soumallyadev07

**Prior experience:**
- [x] HTML
- [x] CSS
- [ ] SCSS
- [ ] Tailwind CSS
- [x] JavaScript
- [ ] TypeScript
- [ ] Vue.js
- [ ] Nuxt.js
- [ ] NestJS
- [x] Firebase
- [ ] Prisma
- [ ] GraphQL
- [ ] PostgreSQL
- [ ] Go lang

---

**Proposal type:**
- [ ] Existing feature request
- [x] New feature proposal
- [ ] Refactor existing feature

**Proposal title:** Graphical Interface for API Response Visualization

**Problem description:** When the response data is large and consists of multiple dictionaries and sub-dictionaries, it is very difficult to understand and use the data. Due to this, the subsequent development gets delayed or the code gets complicated to match the complexity of the response object.

**Proposal description:** Since we can't change the response object as it will corrupt the data our user is trying to get or understand, we can try and make a beautiful visualization of the response using a hierarchical collapsable drawer-like graphical interface so that the user can minimize the necessary parts of the response and focus only on the parts important to them, this will also help them easily understand the relationship between different data points and their parenthood or children.

**Proposal roadmap:** 
1. Create a basic visualization structure using HTML, CSS & JavaScript
2. Add animations & drawer-like features so that they are collapsable
3. Back-end for the visualization - take the response object and populate the HTML fields using JavaScript
4. Testing 
5. Integration with the portal

**Additional context**
Adding a visual component might help developers to accelerate their development and not spend much time making sense out of their data.
![74-2](https://user-images.githubusercontent.com/55331884/118976767-0202c880-b993-11eb-8696-d7f9070bd140.png)

**Comment by liyasthomas** [comment]
Hi @soumallyadev07,
I am afraid we already have such a similar component in the response section which allows me to navigate through the request object.
 
<img width="1440" alt="Screenshot 2021-05-21 at 10 11 30 AM" src="https://user-images.githubusercontent.com/10395817/119083086-2c18b280-b9b4-11eb-846d-b942cb7e029c.png">

**Reply by soumallyadev07** [reply]
Hey! Thanks!
Should I modify my proposal to make improvements in the existing features and make them more interactive & graphical or try to think something else?

**Reply by liyasthomas** [reply]
Yeah you better modify this proposal with another feature that suits you well.

**Reply by KaKi87** [reply]
Hi,
Where is that feature ?
Thanks
